src/SyncFrom/Manager.php

```diff
@@ -0,0 +1,209 @@
+      $options = [
+        'limit' => 100,
+      ];
+    }
+    $entities = $this->clientAdapter->list($options);
+
+    // Finally, sync the entities.
+    foreach ($entities as $entity) {
+      $this->sync($entity);
+    }
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  public function sync($remote_entity) {
+    // Dispatch an event to define which Drupal entity and ID to sync this
+    // remote entity with.
+    $event = new EntityMappingEvent($remote_entity, []);
+    $this->eventDispatcher->dispatch(
+      EntityMappingEvent::EVENT_NAME,
+      $event
+    );
+    $entity_mapping = $event->getEntityMapping();
+    if (empty($entity_mapping['type'])) {
+      $message = 'There is no entity mapping information for this entity.';
+      $this->logger->error($message);
+      throw new \RuntimeException($message);
+    }
+
+    // Now, dispatch another event to define which Drupal entity fields will be
+    // synced to which remote entity fields.
+    $event = new FieldMappingEvent($remote_entity, []);
+    $this->eventDispatcher->dispatch(FieldMappingEvent::EVENT_NAME, $event);
+    $field_mapping = $event->getFieldMapping();
+    if (empty($field_mapping)) {
+      $message = 'There is no field mapping information for this entity.';
+      $this->logger->error($message);
+      throw new \RuntimeException($message);
+    }
+
+    // Now, run another event to do the official syncing.
+    $event = new EntityUpdateEvent(
+      $remote_entity,
+      $entity_mapping,
+      $field_mapping
+    );
+    $this->eventDispatcher->dispatch(
+      EntityUpdateEvent::EVENT_NAME,
+      $event
+    );
+  }
+
+  /**
+   * Check if the application is properly set up for syncing the given entity.
+   *
+   * @param string $entity_type
+   *   Then entity type we're trying to sync.
+   * @param string $bundle
+   *   The bundle of the entity type.
+   *
+   * @return bool
+   *   Returns TRUE if the app is ready for syncing the entity.
+   */
+  protected function appReady($entity_type, $bundle) {
+    $app_ready = TRUE;
+
+    // First check if the entity type bundle has the required fields.
+    $required_fields = [
+      'sync_remote_id',
+      'sync_changed',
+    ];
+    foreach ($required_fields as $field_name) {
+      if (!$this->bundleHasField(
+        $entity_type,
+        $bundle,
+        $field_name)
+      ) {
+        $app_ready = FALSE;
+      }
+    }
+
+    return $app_ready;
+  }
+
+  /**
+   * Checks if an entity bundle has a specific field.
+   *
+   * @param string $entity_type
+   *   Then entity type.
+   * @param string $bundle
+   *   The bundle of the entity type.
+   * @param string $field_name
+   *   The name of the field to check.
+   *
+   * @return bool
+   *   Returns TRUE if the field exists in the entity bundle.
+   */
+  protected function bundleHasField($entity_type, $bundle, $field_name) {
+    $field_storage = FieldStorageConfig::loadByName(
+      $entity_type,
+      $field_name
+    );
```

Comment:
**CHANGE REQUEST**

Please inject the corresponding service; look at the `loadByName` function itself.

Reply:
This is a ConfigEntityType and I've looked everywhere for how to inject it as a service and I just don't think we can, including, looking for the loadByName function.